### PR TITLE
Template logic for updates and attrs moved to resource classes

### PR DESCRIPTION
**Pull request by pinzon**
Work in progress to make the template_deployer.py leaner.

### CODE REVIEW

File: localstack/services/cloudformation/models/s3.py [hunk 1]
```diff
@@ -215,7 +215,18 @@ def get_cfn_attribute(self, attribute_name):
         if attribute_name in ["DomainName", "RegionalDomainName"]:
             bucket_name = self._get_bucket_name()
             return "%s.%s" % (bucket_name, S3_VIRTUAL_HOSTNAME)
+
+        if attribute_name == "WebsiteURL":
+            bucket_name = self.props.get("BucketName")
+            return f"http://{bucket_name}.{S3_STATIC_WEBSITE_HOSTNAME}"
```
Comment (thrau) [inline localstack/services/cloudformation/models/s3.py:221]: nit: technically the s3 bucket could also have an https URL.

**Review by thrau — APPROVED** [review]
s3 changes LGTM!

**Review by dominikschubert — APPROVED** [review]
LGTM! :rocket: 

The only bad thing now is that the low-hanging fruits in template_deployer are gone :grimacing:

**Comment by thrau** [comment]
> LGTM! rocket
> 
> The only bad thing now is that the low-hanging fruits in template_deployer are gone grimacing

there worse problems to have :D

**Pull request merged**